File: 2574-left-and-right-sum-differences/2574-left-and-right-sum-differences.swift
class Solution {
    func leftRightDifference(_ nums: [Int]) -> [Int] {
        var result: Array<Int> = []
        var leftSum: Array<Int> = [0]
        var rightSum: Array<Int> = [0]
        
        for i in 1 ..< nums.count {
            leftSum.append(nums[i - 1] + leftSum.last!)
        }
        for i in stride(from: nums.count - 1, to: 0, by: -1) {
            rightSum.append(nums[i] + rightSum.last!)
        }
        
        rightSum.reverse()
        
        for i in 0 ..< leftSum.count {
            result.append(abs(leftSum[i] - rightSum[i]))
        }
        
        return result
    }
}

File: 2574-left-and-right-sum-differences/2574-left-and-right-sum-differences.cs
public class Solution
{
    public int[] LeftRightDifference(int[] nums)
    {
        int[] result = new int[nums.Length];
        int[] leftSum = new int[nums.Length];
        int[] rightSum = new int[nums.Length];

        int i = 0;
        int j = nums.Length - 1;

        while (i < nums.Length || j >= 0)
        {
            if (i == 0)
            {
                leftSum[i] = 0;
            }
            else
            {
                leftSum[i] = leftSum[i - 1] + nums[i - 1];
            }

            if (j == nums.Length - 1)
            {
                rightSum[j] = 0;
            }
            else
            {
                rightSum[j] = rightSum[j + 1] + nums[j + 1];
            }

            i++;
            j--;
        }

        for (int k = 0; k < nums.Length; k++)
        {
            result[k] = Math.Abs(leftSum[k] - rightSum[k]);
        }

        return result;
    }
}